npm/vite-dev-server/src/resolveConfig.ts

```diff
@@ -128,4 +128,12 @@ function makeCypressViteConfig (config: ViteDevServerConfig, vite: Vite): Inline
       CypressSourcemap(config, vite),
     ],
   }
+
+  if (!vite.version || !semverGte(vite.version, '4.2.0')) {
+    // We are using Vite 4.2.0+, so `incremental` doesn't exist in the types
+    // @ts-expect-error
+    viteConfig.optimizeDeps!.esbuildOptions!.incremental = true
```

Comment:
I think if we reverse the logic we don't need the `!` assertions or the `@ts-expect`. Have `incremental: true` in the base config, then `delete` if version > 4.2.0

`delete viteConfig.optimizeDeps?.esbuildOptions?.incremental`